src/__tests__/resources/assets.integration.spec.ts

```diff
@@ -108,10 +109,31 @@ describe('Asset integration test', () => {
       ...childArr,
     ]);
     expect(createdAssets.length).toBe(1001);
-    await client.assets.delete(
-      createdAssets.slice(1).map(asset => ({ id: asset.id }))
-    );
-    await client.assets.delete([{ id: createdAssets[0].id }]);
+    await client.assets.delete([{ id: createdAssets[0].id }], {
+      recursive: true,
+    }); // only need to delete the root asset
+  });
+
+  test('fail to delete a root asset with children', async () => {
+    const newRootAsset = {
+      ...rootAsset,
+      externalId: 'test-root' + randomInt(),
+    };
+    const newChildAsset = {
+      ...childAsset,
+      parentExternalId: newRootAsset.externalId,
+    };
+    await client.assets.create([newRootAsset, newChildAsset]);
+    await sleepPromise(2000);
+    const prom = client.assets.delete([
+      { externalId: newRootAsset.externalId },
+    ]);
+    expect(prom).rejects.toThrow();
```

Comment:
don't you want to check error message here?

Reply:
Fixed it now